aviary/utils/propeller_map_conversion.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/python
+
+import argparse
+import getpass
+import itertools
+import numpy as np
+from datetime import datetime
+from enum import Enum, auto
+from pathlib import Path
+
+from aviary.api import NamedValues
+from aviary.utils.csv_data_file import write_data_file
+from aviary.utils.functions import get_path
+from aviary.subsystems.propulsion.utils import default_units
+
+
+class PropMapType(Enum):
+    GASP = 'GASP'
+
+    def __str__(self):
+        return self.value
+
+
+class PropModelVariables(Enum):
+    '''
+    Define constants that map to supported variable names in an propeller model.
+    '''
+    MACH = auto()
+    CP = auto()  # power coefficient
+    CT = auto()  # thrust coefficient
+    J = auto()  # advanced ratio
+
+
+default_units = {
+    PropModelVariables.MACH: 'unitless',
+    PropModelVariables.CP: 'unitless',
+    PropModelVariables.CT: 'unitless',
+    PropModelVariables.J: 'unitless',
+}
+
+MACH = PropModelVariables.MACH
+CP = PropModelVariables.CP
+CT = PropModelVariables.CT
+J = PropModelVariables.J
+
+gasp_keys = [MACH, CP, CT, J]
+
+header_names = {
+    MACH: 'Mach_Number',
+    CP: 'CP',
+    CT: 'CT',
+    J: 'J',
+}
```

Comment:
do we need both of these? could we make the values of the PropModelVariables be these strings and use `.keys()` instead of a separate gasp_keys list?

Reply:
I thought about it but I wasn't sure if we would add other conversion cases. So, I decided to follow `engine_deck_conversion.py`. If we will only convert GASP table, then yes, I can simply the code at several places.

Reply:
I think we should combine the header_names into the PropModelVariables as the values, but we can leave the gasp_keys in case other prop models need to get converted.

Reply:
`gasp_keys` is not used and is removed.

Reply:
Now, header_names dict is created from PropModelVariables.